fe/fe-core/src/main/java/org/apache/doris/analysis/SelectStmt.java

```diff
@@ -851,6 +852,11 @@ private void whereClauseRewrite() {
             } else {
                 whereClause = new BoolLiteral(true);
             }
+        } else if (whereClause instanceof CaseExpr
+                && (((CaseExpr) whereClause).getChild(1) instanceof IntLiteral)
+                && (((CaseExpr) whereClause).getChild(2) instanceof IntLiteral)) {
+            whereClause = new CastExpr(TypeDef.create(PrimitiveType.BOOLEAN), whereClause);
+            whereClause.setType(Type.BOOLEAN);
```

Comment:
all whereClause should boolean type, so do not only process case when

Reply:
done